### PR TITLE
NXP-28450: allow configuration of Signature Algorithm for SAML (10.10)

### PR DESCRIPTION
**Pull request by efge**
https://qa2.nuxeo.org/jenkins/job/TestAndPush/job/ondemand-testandpush-fguillaume-10.10/143/

### CODE REVIEW

File: nuxeo-services/login/nuxeo-platform-login-saml2/src/main/java/org/nuxeo/ecm/platform/auth/saml/SAMLAuthenticationProvider.java [hunk 1]
```diff
@@ -254,6 +258,27 @@ private void addProfile(AbstractSAMLProfile profile) {
...
+    private static final String SIGNATURE_ALGORITHM = "SignatureAlgorithm";
+
+    private static final String DIGEST_ALGORITHM = "DigestAlgorithm";
+
+    private void initializeSecurity(Map<String, String> parameters) {
+        BasicSecurityConfiguration config = (BasicSecurityConfiguration) Configuration.getGlobalSecurityConfiguration();
+        for (Entry<String, String> es : parameters.entrySet()) {
+            String key = es.getKey();
+            String value = es.getValue();
+            if (key.startsWith(SIGNATURE_ALGORITHM)) { //
```
Comment (troger) [inline nuxeo-services/login/nuxeo-platform-login-saml2/src/main/java/org/nuxeo/ecm/platform/auth/saml/SAMLAuthenticationProvider.java:270]:
```suggestion
            if (key.startsWith(SIGNATURE_ALGORITHM)) {
```
or missing comment maybe?

Comment (efge) [reply nuxeo-services/login/nuxeo-platform-login-saml2/src/main/java/org/nuxeo/ecm/platform/auth/saml/SAMLAuthenticationProvider.java]: Unfinished comment but that's not really needed the code is clear enough now.

**Comment by efge** [comment]
(force push to remove empty comment)

**Comment by efge** [comment]
(rebase)

**Review by troger — APPROVED** [review]

**Comment by efge** [comment]
Fix validated by customer

**Review by ataillefer — APPROVED** [review]

**Pull request merged**